src/commands/dependency/reset.ts

```diff
@@ -24,17 +34,90 @@ export default class Reset extends BaseCommand<typeof Reset> {
   static override examples = ["<%= config.bin %> <%= command.id %>"];
   static override flags = {
     ...baseFlags,
+    ...GLOBAL_FLAG,
+    all: Flags.boolean({
+      description: "Remove all dependencies, not only recommended ones",
+    }),
   };
   async run(): Promise<void> {
-    const { fluenceConfig } = await initCli(
+    const { maybeFluenceConfig, flags } = await initCli(
       this,
-      await this.parse(Reset),
-      true
+      await this.parse(Reset)
     );
 
-    delete fluenceConfig.dependencies;
-    await fluenceConfig.$commit();
+    if (flags.global) {
```

Comment:
Let's move this into a separate function. So that there are two functions: `resetGlobal` and `resetLocal`.

Single "do-it-all" function is very hard to understand.

Reply:
I think this is just unecessary and inconcistent. -g or --global flag is used in other commands as well and I don't think it is that hard wrap your head around. Having one more command here will not decrese complexity in any way in my opinion - it will just spread the complexity around a little bit

Reply:
also all this dependency management I expect to be more for advanced users, especially untill we have proper dependency management on aqua side. So I expect users to mostly use the stuff we provided with the versions we provided. I unfortunately don't expect much publishing of their own libraries for now that they will themselves use